### PR TITLE
core: refactor datachannel creation

### PR DESCRIPTION
**Pull request by saghul**
Fixes: https://github.com/react-native-webrtc/react-native-webrtc/issues/962

### CODE REVIEW

**Comment by ibc** [comment]
Nice. Is a NPM release coming?

**Comment by saghul** [comment]
Maybe tomorrow.

**Comment by ibc** [comment]
Oh, thanks a lot!

**Comment by saghul** [comment]
Done. 1.89.2 is out: https://github.com/react-native-webrtc/react-native-webrtc/releases/tag/1.89.2 also on npm.

File: ios/RCTWebRTC/DataChannelWrapper.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+- (void)dataChannel:(nonnull RTCDataChannel *)dataChannel didReceiveMessageWithBuffer:(nonnull RTCDataBuffer *)buffer {
+    if (_delegate) {
+        [_delegate dataChannel:self didReceiveMessageWithBuffer:buffer];
+    }
+}
+
+- (void)dataChannelDidChangeState:(nonnull RTCDataChannel *)dataChannel {
+    if (_delegate) {
```
Comment (xilin) [inline ios/RCTWebRTC/DataChannelWrapper.m:32]: 💯  great idea of adding this wrapper

File: ios/RCTWebRTC/WebRTCModule+RTCDataChannel.m [hunk 2]
```diff
@@ -1,82 +1,99 @@
...
-}
+@implementation WebRTCModule (RTCDataChannel)
 
-/**
- * webrtc will update channelId internally:
- * https://github.com/jitsi/webrtc/blob/f3736ed3d8d4e6660ad8b82a8deff7aca8389e5c/pc/data_channel.cc#L39
- * https://github.com/jitsi/webrtc/blob/f3736ed3d8d4e6660ad8b82a8deff7aca8389e5c/pc/data_channel.cc#L58
- * so with this property we can keep channelId passed from JS side
+/*
+ * Thuis methos is implemented synchronously since we need to create the DataChannel on the spot
```
Comment (xilin) [inline ios/RCTWebRTC/WebRTCModule+RTCDataChannel.m:15]: nit: typo here `This method is ...`

File: ios/RCTWebRTC/DataChannelWrapper.m (hunk 1 above)
Comment (saghul) [reply ios/RCTWebRTC/DataChannelWrapper.m:32]: Thanks for taking a look!

**Pull request merged**